debian/changelog

```diff
@@ -1,9 +1,12 @@
-cloud-init (23.4-0ubuntu1~23.10.2) UNRELEASED; urgency=medium
+cloud-init (23.4.1-0ubuntu1~23.10.1) mantic; urgency=medium
 
   * d/p/retain-apt-pre-deb822.patch:
     - Disable apt source list generation with DEB822 style
+  * Upstream snapshot based on 23.4.1. (LP: #2045582).
```

Comment:
The patch here in debian/patches/do-not-block-user-login.patch still has a redacted patch due to 46992d3c3cbdbdf47d119f3dcba2048f82720c97. This was fixed in other series, but ignored in mantic for some reason. I've cherry-picked a functional change 8e66cec4a41370914132308959eeebe38a5cbf25 from focal's branch to correct this issue in mantic-23.4.x branch. We need this in upstream/ubuntu/mantic as well so it's not forgotten on next release. I'll put up a separate PR to sync that fix into `upstream/ubuntu/mantic` now.   Also debian changelog doesn't note the refreshed quilt patches, or the dropped cherry-pick(cpick) files. I've added a changelog delta to reflect this content.

1. first diff for fixing/reverting the quilt patch to avoid redacting the patchset
```diff
--- a/debian/patches/do-not-block-user-login.patch
+++ b/debian/patches/do-not-block-user-login.patch
@@ -16,3 +16,13 @@ This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
  Wants=network-online.target cloud-config.target
  ConditionPathExists=!/etc/cloud/cloud-init.disabled
  ConditionKernelCommandLine=!cloud-init=disabled
+--- a/systemd/cloud-init.service.tmpl
++++ b/systemd/cloud-init.service.tmpl
+@@ -38,6 +38,7 @@ Conflicts=shutdown.target
+ Before=shutdown.target
+ Conflicts=shutdown.target
+ {% endif %}
++Before=systemd-user-sessions.service
+ ConditionPathExists=!/etc/cloud/cloud-init.disabled
+ ConditionKernelCommandLine=!cloud-init=disabled
+ ConditionEnvironment=!KERNEL_CMDLINE=cloud-init=disabled
```

2. second diff for the updated debian/changelog capturing both the refreshed patches, the dropped cherry-picks and the fixed do-not-block-user-login.patch
```diff
--- a/debian/changelog
+++ b/debian/changelog
@@ -2,6 +2,11 @@ cloud-init (23.4.1-0ubuntu1~23.10.1) mantic; urgency=medium
 
   * d/p/retain-apt-pre-deb822.patch:
     - Disable apt source list generation with DEB822 style
+  * d/p/do-not-block-user-login.patch:
+    - revert redacted patch content introduced in 23.4-0
+  * refresh patches:
+    - d/p/status-do-not-remove-duplicated-data.patch
+  * d/changelog: amend 23.4-0 refresh patches and dropped cherry-picks entry
   * Upstream snapshot based on 23.4.1. (LP: #2045582).
     List of changes from upstream can be found at
     https://raw.githubusercontent.com/canonical/cloud-init/23.4.1/ChangeLog
@@ -13,6 +18,14 @@ cloud-init (23.4-0ubuntu1~23.10.1) mantic; urgency=medium
   * d/p/status-do-not-remove-duplicated-data.patch:
     - Revert behavior downstream, leave duplicate data
   * d/control: add python3-apt as Recommends to read APT config from apt_pkg
+  * refresh patches:
+    - d/p/do-not-block-user-login.patch
+  * drop the following cherry-picks now included:
+    - cpick-0d9f149a-Pytestify-apt-config-test-modules-4424
+    - cpick-5023e9f9-Refactor-test_apt_source_v1.py-to-use-pytest-4427
+    - cpick-e9cdd7e3-Install-gnupg-if-gpg-not-found-4431
+    - cpick-015543d3-apt-install-software-properties-common-when-absent-but
+    - cpick-2ab1f340-fix-cc_apt_configure-avoid-unneeded-call-to-apt-install
   * Upstream snapshot based on 23.4. (LP: #2045582).
     List of changes from upstream can be found at
     https://raw.githubusercontent.com/canonical/cloud-init/23.4/ChangeLog
```